### DISCUSSION TITLE
Unknown Issuer Error

### DISCUSSION THREAD

**Discussion by akash3gupta** [svix/svix-webhooks#760] [General]
I deployed svix server in my local machine by using your docker compose file (docker compose is using "svix/svix-server" image). I did not get any issue till this step.
I registered a app, some events, and endpoints also. When I am creating a message, it is trying to call registered endpoints, but It is not able to hit endpoints properly. I am getting some Unkown Issuer error or some certificate errors.
What do I need to do to reslove this issue?

**Comment by jaymell** [comment]
Hi, can you paste the actual error you're seeing? 

I'm not sure offhand if this is a client- or server-side issue, so I'm wondering, if you haven't already, if you can try to create an endpoint using a URL generated by [Svix Play](https://play.svix.com/) to see if this also fails.

**Reply by akash3gupta** [reply]
Now, I am getting below error:
<img width="571" alt="image" src="https://user-images.githubusercontent.com/21129271/212638488-99cfd6f9-9876-4f23-b4f9-9d2f8de5729d.png">

**Comment by akash3gupta** [comment]
This is the UnknowIssuer error screenshot. Please check it and give me some suggestions what thing I am doing wrong here
<img width="626" alt="image" src="https://user-images.githubusercontent.com/21129271/212678746-291534dd-716c-41db-9cc4-82b366fe2efb.png">

**Reply by akash3gupta** [reply]
Even for my locally deployed API, I am getting same error:
<img width="593" alt="image" src="https://user-images.githubusercontent.com/21129271/212685896-18ab3b9c-08f9-401e-b9bf-5bbdcab2ce98.png">
@tasn Can you help me here? What is the reason behind it?

**Reply by tasn** [reply]
You are trying to connect to an `https` server with a self-signed certificate and the svix cli (or svix, or whatever) is rightfully rejecting it because the certificate can't be verified.

It says it in the error: unknown issuer.

**Reply by akash3gupta** [reply]
Thanks for the reply. @tasn  So Can I use svix play for my testing? Because svix play is also throwing error.
![image](https://user-images.githubusercontent.com/21129271/212745802-792bb072-8459-49e8-aae3-d887e7c7a6ae.png)

**Reply by akash3gupta** [reply]
Hi @tasn  I have resolved the certificate issue. Now I am getting this issue.
<img width="617" alt="image" src="https://user-images.githubusercontent.com/21129271/212844669-aea0613b-ff1a-493c-bbc9-d49b1d6291de.png">

**Reply by tasn** [reply]
Try again?

**Reply by akash3gupta** [reply]
I tried lot of time.

**Reply by tasn** [reply]
It could be a firewall on your network or something. Are you even able to send a webhook anywhere?

**Comment by cheleswer** [comment]
How can I send the event to the endpoints which are using self-signed certificates? Is this possible in SVIX-SERVER? I am running a POC for webhooks and sending events to the internal system which may not have publicly verified certifications.

**Reply by tasn** [reply]
Hm... The easiest way is probably adding the private CA certificate (what you're self-signing with) to the Docker image's certificate store. Though why use self-signed certs instead of e.g. `http`?

**Reply by cheleswer** [reply]
That's because our product is mostly deployed in clients' close environments and sometimes they don't really want to use the public certificates for their internal communication but want to use SSL (always https). My requirement is to keep adding new self-signed certs as and when needed for the different clients, I am not sure how the docker images certificate store will work though.

**Reply by tasn** [reply]
Can you open a ticket to support allowing ignoring certificate authenticity? We can discuss it further then, and maybe take action.

**Reply by cheleswer** [reply]
sure,  created below ticket
Allowing ignoring certificate authenticity #1030

**Reply by tasn** [reply]
Thanks!
